tests/uv/manager/test_measurementmanager.py

```diff
@@ -0,0 +1,248 @@
+#!/usr/bin/python
+#
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests of manager functions."""
+
+import uv.reporter as r
+import uv.manager as m
+
+
+def test_manager_setup():
+  """Tests whether the MeasurementManager() object is setup
+  without error"""
+
+  reporter = r.MemoryReporter().stepped()
+  static_state = {}
+
+  test_manager = m.MeasurementManager(static_state, reporter)
+
+
+def test_add_measurement():
+  """Tests that measurements can be added to a MeasurementManager
+  object properly """
+
+  reporter = r.MemoryReporter().stepped()
+  static_state = {}
+
+  test_manager = m.MeasurementManager(static_state, reporter)
+
+  assert len(test_manager.measurements) == 0
+
+  test_manager.add_measurement({
+      'name': 'L2',
+      'interval': 10,
+      'function': lambda state: 0
+  })
+
+  assert len(test_manager.measurements) == 1
+  assert 'L2' in test_manager.measurements.keys()
+  assert test_manager.measurements['L2']['interval'] == 10
+  assert test_manager.measurements['L2']['function'](static_state) == 0
+
+
+def test_measurement_invervals():
+  """Tests that measurements occur at the specified intervals"""
+
+  data_store = {}
+  reporter = r.MemoryReporter(data_store).stepped()
+  static_state = {}
+
+  test_manager = m.MeasurementManager(static_state, reporter)
+
+  test_intervals = [10, 13, 17, 20]
+  for interval in test_intervals:
+    test_manager.add_measurement({
+        'name': f'Int_{interval}',
+        'interval': interval,
+        'function': lambda state: 0
+    })
+
+  TEST_STEPS = 100
+  for i in range(TEST_STEPS):
+    test_manager.process(i, {})
+
+  # check data_store
+  assert data_store == {
+      'Int_10': [{
```

Comment:
can I get you to tighten this up? something like

```python
{
  'Int_10': [{'value': 0, 'step': s} for s in range(0, 100, 10)],
  'Int_13': [{'value': 0, 'step': s} for s in range(0, 100, 13)],
  'Int_17': [{'value': 0, 'step': s} for s in range(0, 100, 17)],
  'Int_20': [{'value': 0, 'step': s} for s in range(0, 100, 20)],
```

you can probably see how to tighten it up a little more. It makes the tests much easier to read, and easier to convert to a generative testing style later.

Reply:
Ah ok, yep will do.  For some reason, I thought that the convention was to be more verbose in tests, but I don't know where I got that idea from.